packages/nuxt/src/core/runtime/nitro/error.ts

```diff
@@ -6,6 +6,7 @@ import { useNitroApp, useRuntimeConfig } from '#internal/nitro'
 import { isJsonRequest, normalizeError } from '#internal/nitro/utils'
 
 export default <NitroErrorHandler> async function errorhandler (error: H3Error, event) {
+  if (event.handled) { return }
```

Comment:
Since we might have hooks and logs, this check around `res.end` is probably enough and can be removed from here

Reply:
happy to move it later when we have hooks - but I think we do need it currently as calling `setResponseHeader` will also throw if the event has already been sent back, right?

Reply:
Hmm makes sense to also guard it but it only sets a map value (but not sends it without `res.end/send/sendHeaders)